lib/datasource/galaxy-collection/index.ts

```diff
@@ -0,0 +1,156 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import * as packageCache from '../../util/cache/package';
+import { Http } from '../../util/http';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export const id = 'galaxy-collection';
+export const defaultRegistryUrls = ['https://galaxy.ansible.com/'];
+export const customRegistrySupport = false;
+
+const http = new Http(id);
+
+interface BaseProjectResult {
+  versions_url: string;
+  deprecated: boolean;
+  latest_version: {
+    version: string;
+  };
+}
+
+interface VersionsProjectResult {
+  results: Versions[];
+}
+
+interface VersionsDetailResult {
+  download_url: string;
+  artifact: {
+    filename: string;
+    size: bigint;
+    sha256: string;
+  };
+  metadata: {
+    homepage: string;
+    tags: Record<string, string>;
+    dependencies: Record<string, string>;
+    repository: string;
+  };
+}
+
+interface Versions {
+  version: string;
+  href: string;
+}
+
+export async function getReleases({
+  lookupName,
+  registryUrl,
+}: GetReleasesConfig): Promise<ReleaseResult | null> {
+  const cacheNamespace = 'datasource-galaxy-collection';
+  const cacheKey = lookupName;
+  const cachedResult = await packageCache.get<ReleaseResult>(
+    cacheNamespace,
+    cacheKey
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const lookUp = lookupName.split('.');
+  const namespace = lookUp[0];
+  const projectName = lookUp[1];
+
+  const baseUrl = `${registryUrl}api/v2/collections/${namespace}/${projectName}/`;
+
+  try {
+    const baseUrlResponse: any = await http.get(baseUrl);
+    if (!baseUrlResponse || !baseUrlResponse.body) {
+      logger.warn(
+        { dependency: lookupName },
+        `Received invalid data from ${baseUrl}`
+      );
+      return null;
+    }
+
+    const baseProject: BaseProjectResult = JSON.parse(baseUrlResponse.body);
+
+    const versionsUrl = `${baseUrl}versions/`;
+    const versionsUrlResponse: any = await http.get(versionsUrl);
+    if (!versionsUrlResponse || !versionsUrlResponse.body) {
+      logger.warn(
+        { dependency: lookupName },
+        `Received invalid data from ${baseUrl}versions/`
+      );
+      return null;
+    }
+
+    const versionsProject: VersionsProjectResult = JSON.parse(
+      versionsUrlResponse.body
+    );
+
+    const releases: Release[] = versionsProject.results.map((value) => {
+      const release: Release = {
+        version: value.version,
+        isDeprecated: baseProject.deprecated,
+      };
+      return release;
+    });
+
+    let newestVersionDetails: VersionsDetailResult;
+    // asynchronously get release details
+    const enrichedReleases: Release[] = await Promise.all(
+      releases.map(async (basicRelease) => {
```

Comment:
I think this needs to be rate limited, otherwise we will open hundreds of parallel http requests.

Sample:
https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/datasource/nuget/v3.ts#L142